### DISCUSSION TITLE
Taudem stream shape result has not prj file.

### DISCUSSION THREAD

**Discussion by mytudousi** [DotSpatial/DotSpatial#1499] [General]
Plugins.Taudem.Hydrology.cs file line 1729 is 
DataManagement.TryCopy(Path.ChangeExtension(demGridPath, ".prj"), Path.ChangeExtension(streamShapeResultPath, ".prj"));

Because after StreamNet.exe  execution,  the result of stream shape file has not prj file.  So the purpose of this code is set .prj file to stream shape file.
For example, our dem file name is A.tiff, the dem file must has A.prj file.  if the A.prj file doesn't exist, the stream shape file will not has .prj file. 
But, In general, raster file such as .tif .tiff .img , etc，has not .prj file.

So, I want to change the code. first, I will determine if the dem grid file contains the .prj file? Yes，use current code. No, do the following.
1、Use the OpenFile function of Raster class, open dem grid file, get IRaster object;
2、Use IRaster.Projection, get ProjectionInfo object;
3、Call IRaster.Dispose() release resource;
4、Call ProjectionInfo.SaveAs() function, save project info as streamShapeResultPath.prj file.

I wonder if this is OK?

**Comment by sindizzy** [comment]
Sounds reasonable. If I were you I would write a pull request and get it reviewed.

**Reply by mytudousi** [reply]
OK，I'll do it in the next two days。

**Comment by jany-tenaj** [comment]
fixed by #1502
